src/bpftrace.cpp

```diff
@@ -1956,20 +1956,24 @@ std::set<std::string> BPFtrace::list_modules(const ast::ASTContext &ctx)
       auto probe_type = probetype(ap->provider);
       if (probe_type == ProbeType::fentry || probe_type == ProbeType::fexit ||
           probe_type == ProbeType::rawtracepoint ||
-          ((probe_type == ProbeType::kprobe ||
-            probe_type == ProbeType::kretprobe) &&
-           !ap->target.empty())) {
-        if (ap->expansion != ast::ExpansionType::NONE) {
```

Comment:
Do we know what this `ast::ExpansionType::NONE` was doing?

Reply:
I think the idea was to avoid making a call to ProbeMatcher if the probe didn't need expansion (as presumably there would be no additional matches). But to fix this bug, it was simpler to put everything on the ProbeMatcher codepath rather than carve out further exceptions in the name of performance.

Reply:
kk - makes sense.